Add odk-central to POSM

thenav56 opened the pull request:
### Changes
- Add [ODK Central](https://github.com/opendatakit/central) to posm at `odk-central.posm.io`.
- Repository used: (Using branch `posm` for all repository)
   - https://github.com/posm/central/tree/posm
   - https://github.com/posm/central-frontend/tree/posm
   - https://github.com/posm/central-backend/tree/posm
- Diff available here https://github.com/opendatakit/central/compare/master...posm:posm
- Creating initial user with credential `posm@posm.io:awesomeposm`.

### Feedback Fixes
- central diff https://github.com/posm/central/commit/e25606b52510a9b843d66c43d1737c19cef45279
- Add pyxform as a dependent of the odk service (backend). https://github.com/posm/posm/issues/364
- Using local directory for docker volume to preserve data from posm-build. https://github.com/posm/posm/issues/363

thenav56 committed:
Add odk-central to POSM

danbjoseph commented:
i asked on the [ODK slack](https://getodk.slack.com/join/shared_invite/zt-ef9bkpjl-3ISyqs4HnFi0mIuLlJgkQw#/) in the `central-code` channel about the above issues and they mentioned...
> have y’all checked the obvious like making sure https://github.com/getodk/pyxform-http is running and plumbed so Central can talk to it? If you’re off the standard install you’ll need to do that yourselves.

and
> One thing we’d likely like to have is the ability for admins to reset user passwords. That could be something to contribute so you don’t need to fork.

and
> yeah there are more components now than just frontend/backend, and soon to be even more.

thenav56 committed:
Fix

- odk central admin user not working #363
- error when uploading XLSForm to ODK Central #364
- Allow http for odk-collect

danbjoseph commented:
since "there are more components now than just frontend/backend, and soon to be even more" is there a way we can try and do something closer to the standard install? instead of installing the components individually?

the password functionality would be something we could talk to the ODK team about, and then submit as a PR.

thenav56 commented:
@danbjoseph 

> since "there are more components now than just frontend/backend, and soon to be even more" is there a way we can try and do something closer to the standard install? instead of installing the components individually?

We followed the instructions provided by the central [documentation](https://docs.getodk.org/central-install-digital-ocean/#installing-odk-central)
The missing part was the [here](https://github.com/posm/central/blob/master/docker-compose.yml#L24-L26) where `pyxform` was not specified. We modified the [service](https://github.com/posm/central/blob/master/files/docker-compose%40.service) file to only start the service (backend) as the frontend was served by posm nginx to posm-build [service file](https://github.com/posm/posm-build/pull/126/files#diff-3275847c085f0a1b6c72f9749670e319).
After adding the pyxform all should work as expected.
In the future if they add/remove components we only have to copy them from their docker-compose.yml.

> the password functionality would be something we could talk to the ODK team about, and then submit as a PR.

Right now there is a extra step which allows admin to share password reset URL with user or use themselves. This was preferable as little changes were only needed allowing password modification by actors. 
Talking to ODK team sounds good. cc: @bewakes

danbjoseph commented:
do you want to join the ODK Slack or should i try and setup a call?

bewakes commented:
we joined the ODK slack. Let us know if you want to have a call.

danbjoseph commented:
apologies, this is better discussed on the ODK forum not the Slack workspace
https://forum.getodk.org/t/central-password-reset-without-email/27076
